manta-js/package/sdk.ts

```diff
@@ -0,0 +1,604 @@
+        const wasm = await import('manta-wasm-wallet');
+        const wasmSigner = new wasm.Signer(SIGNER_URL);
+        const wasmApiConfig = new ApiConfig(
+            maxReceiversPullSize, maxSendersPullSize, pullCallback, errorCallback
+        );
+        const wasmApi = new Api(api,wasmApiConfig);
+        const wasmLedger = new wasm.PolkadotJsLedger(wasmApi);
+        const wasmWallet = new wasm.Wallet(wasmLedger, wasmSigner);
+        return {
+            wasm,
+            wasmWallet,
+            wasmApi
+        }
+    }
+
+    /// Sets the polkadot Signer to `polkadotSigner` and polkadot signing address to `polkadotAddress`.
+    private async setPolkadotSigner(polkadotSigner: Signer, polkadotAddress:Address): Promise<void> {
+        this.wasmApi.setExternalAccountSigner(polkadotAddress);
+        this.api.setSigner(polkadotSigner);
+    }
+
+    /// Returns the corresponding blockchain connection URL from Environment
+    /// and Network values.
+    private static envUrl(env: Environment, network: Network): string {
+        var url = config.NETWORKS[network].ws_local;
+        if(env == Environment.Production) {
+            url = config.NETWORKS[network].ws;
+        }
+        return url;
+    }
+
+
+    /// Builds the "ToPrivate" transaction in JSON format to be signed.
+    private async toPrivateFungible(assetId: BN, amount: BN): Promise<any> {
+        try {
+            const assetIdArray = Array.from(this.assetIdToUInt8Array(assetId));
+            const txJson = `{ "ToPrivate": { "id": [${assetIdArray}], "value": ${amount.toString()} }}`;
+            const transaction = this.wasm.Transaction.from_string(txJson);
+            return transaction;
+        } catch (error) {
+            console.error('Unable to build "To Private" Transaction.',error);
+        }
+    }
+
+    /// private transfer transaction
+    private async privateFungibleTransfer(assetId: BN, amount: BN, toPrivateAddress: Address): Promise<any> {
+        try {
+            const addressJson = this.convertPrivateAddressToJson(toPrivateAddress);
+            const assetIdArray = Array.from(this.assetIdToUInt8Array(assetId));
+            const txJson = `{ "PrivateTransfer": [{ "id": [${assetIdArray}], "value": ${amount.toString()} }, ${addressJson} ]}`;
+            const transaction = this.wasm.Transaction.from_string(txJson);
+
+            // construct asset metadata json by query api
+            const assetMeta = await this.api.query.assetManager.assetIdMetadata(assetId);
+
+            const json = JSON.stringify(assetMeta.toHuman());
+            const jsonObj = JSON.parse(json);
+            const decimals = jsonObj["metadata"]["decimals"];
+            const symbol = jsonObj["metadata"]["symbol"];
+            const assetMetadataJson = `{ "decimals": ${decimals}, "symbol": "${PRIVATE_ASSET_PREFIX}${symbol}" }`;
```

Comment:
This and the other transfer types seem to duplicate `getAssetMetadata` in utils. I think I was wrong, and we can move `getAssetMetadata` back into `MantaPrivateWallet`, and use it to deduplicate these functions that call `this.api.query.assetManager.assetIdMetadata(assetId);`